docs/proposals/network-policy.md

```diff
@@ -197,6 +254,36 @@ type NetworkPolicyPort struct {
 	Port *intstr.IntOrString `json:"port,omitempty"`
 }
 
+type NetworkCIDRRange struct {
+	// QUESTION: we could just do "Start" and "End" without a prefix to
+	// specify the range.  While this provides a simpler structure, it may
+	// be more error-prone since its less easily validated, and isn't very
+	// nice for IPv6 addresses.
+
+	// An IPv4 or IPv6 address (eg "192.168.1.0" or 2001:db8::) which
+	// together with the Prefix identifies an IP network. If the optional
+	// End property is non-empty, this object selects all IP addresses from
+	// Start through End, inclusive. For example, if Start is "192.168.1.0",
+	// Prefix is "24", and End is "192.168.15.0" this object would select
+	// all IP addresses between 192.168.1.0 and 192.168.15.255 inclusive.
+	// If Start is 192.168.1.0, Prefix is 24, and End is empty, this object
+	// selects only addresses between 192.168.1.0 and 192.168.1.255.
+	Start net.IP `json:"start"`
```

Comment:
My initial take on this is that `NetworkCIDRRange` is more complicated than it needs to be.  CIDR notation already gives us a way to define ranges of IP addresses.  Why do we need a way to define a range of ranges?

Using a range like this makes one specific case (multiple contiguous CIDRs that don't combine to form another CIDR) easier, but makes the mainline case (a single CIDR, or a selection of disjoint CIDRs) no easier and likely more confusing to the user, who is expecting something along the lines of "IP and prefix length" notation rather than "start, prefix, optional end".

My inclination here would be to list out the specific CIDRs that should be allowed, combine CIDRs into a larger CIDR when applicable.